### ISSUE TITLE
model not providing results

### ISSUE THREAD

**Issue by spaboy** [fizyr/keras-retinanet#260]
Hello,
I'm trying to train a model with my dataset through  CSV with a simple test, classifying chia and sesame seeds

I've a mapping data file : 
```
sesami,0
chia,1
```

and a annotations file like this : 
`IMG_20180124_082912.jpg,752,0,1039,181,chia
IMG_20180124_082912.jpg,2018,32,2385,269,sesami
IMG_20180124_082912.jpg,2116,362,2404,668,sesami
IMG_20180124_082912.jpg,2589,153,2937,418,sesami
IMG_20180124_082912.jpg,2626,970,2993,1267,sesami
IMG_20180124_082912.jpg,1480,1647,1805,1940,sesami
IMG_20180124_082912.jpg,2793,2223,3053,2547,sesami
`
Train dataset is 18 images with 194 localized seeds. 
 
When i run train with snapshots i see that the snapshos are being generated but i noticed that those always have the same size is this normla? i tryed to train with 20 steps and 20 epochs. 

Then i run the jupyter notebook with the example using one of my test images, but it doesn't detect nothing.
See attached training and test images : 
![train_image](https://user-images.githubusercontent.com/29656/35581023-f17265ac-05ea-11e8-90ef-c3040b98069a.jpg)
![test_image_sample](https://user-images.githubusercontent.com/29656/35581055-00b8dc44-05eb-11e8-9d16-184929528463.jpg)

We tested FRCNN with same dataset over CNTK and results were Ok, 

Any idea on what i can be doing wrong? how can i debug the model?   Testing with resnet trained model works , so might be something on my training, but i cannot figure it out. 

Thanks

**Comment by hgaiser** [comment]
Hi there,

All snapshots having the same size is completely normal. The weights change, but they are always floats so always stored as 4 bytes.

I have a couple of questions for you:

- what version (git commit hash) of `keras-retinanet` are you on? There were a couple of issues with the data augmentation method that got fixed a few days ago. 
- if you run `keras_retinanet/bin/debug.py --annotations --resize` on your dataset, do the annotations look normal?
- can you share a screenshot of the result of `debug.py` ?
- one possible explanation could be that your images are resized such that the smallest side is 600px, making your objects really small and therefore not detected. Can you try letting the generator generate larger images (by setting the `image_min_side` and `image_max_side` parameters for your generators. They're not exposed as args to the executable, perhaps they should be somehow).

**Comment by spaboy** [comment]
version : 
commit b3f0c055b7a482c0a735c653b4b0134aa399c906 (HEAD -> master, origin/master,
origin/HEAD)

debug.py with --no-resize works! 
![image](https://user-images.githubusercontent.com/29656/35584583-b31af390-05f5-11e8-9f85-58153e483885.png)

so you are right about the resizing problem :+1: 
![debug](https://user-images.githubusercontent.com/29656/35584852-8b6ed82e-05f6-11e8-9234-3b8f0f3f40a3.png)

I'm changing the Generator min_side and max_side to test with full model, so yes would be good to have this as parameters for other users :)

Thanks for your help!!!

**Comment by spaboy** [comment]
trying to make images of 2000 px chaging the image_min_side and image_max_side  causes me OOM , using tensorflow-gpu on a Nvidia Quadro M1000M of 2GB.

Now i'm testing using CPU, but the training is like 20 times slower... so lets see the results tomorrow...

as workarround ... you think that chunking the images should work ? obtaining images of about 800px side

**Comment by spaboy** [comment]
Hello, 

I manage to train the seed custom dataset by VOC with 400x400 images, with 50 epocs and 50 steps the results are up to 0.8 mAP ... so i might need more training.

Now i'm testing with larger images CSV dataset. 

I close the issue, main problem is comming from image Size. 

thanks for your help
